### PR TITLE
Update DAL for vespa bugfixes

### PR DESCRIPTION
**Pull request by olaughter**
DAL releases: https://github.com/climatepolicyradar/data-access/releases

This also means updating fastapi (and starlette) to avoid clashing pydantic versions. FastAPI releases: https://github.com/tiangolo/fastapi/releases

# Description

Please include:
- a summary of the changes
- links to any related issue/ticket
- any additional relevant motivation and context
- details of any dependency updates that are required for this change

## Type of change

Please select the option(s) below that are most relevant:

- [X] Bug fix
- [ ] New feature
- [ ] Breaking change

## How Has This Been Tested?

I'm relying on the existing tests and the review process

## Reviewer Checklist

- [X] The PR represents a single feature (small driveby fixes are also ok)
- [X] The PR includes tests that are sufficient for the level of risk
- [X] The code is sufficiently commented, particularly in hard-to-understand areas
- [X] Any required documentation updates have been made
- [X] Any TODOs added are captured in future tickets
- [X] No FIXMEs remain

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -9,25 +9,22 @@ RUN apt update && \
...
-    > requirements.txt
-# The above final impenetrable sed command allows us to fix an export issue from poetry
-# e.g. we need to replace pydocstyle[pydocstyle[toml]] with pydocstyle[toml]
+    --without-hashes \
+    --format=requirements.txt \
+    --output=requirements.txt
 
 # Install torch-cpu with pip
 RUN pip3 install --no-cache "torch==2.0.0+cpu" "torchvision==0.15.1+cpu" -f https://download.pytorch.org/whl/torch_stable.html
+RUN pip3 install --no-cache "sentence-transformers==2.2.2"
```
[the remark this answers is not in the document]
Comment (olaughter) [reply Dockerfile:27]: It turns out the approach we are taking with the vespa group is not compatible with how we install the backend, I've tried working around it but havent found a way around it yet, I added some detail here: https://github.com/climatepolicyradar/navigator-backend/pull/189/commits/eeee5012db5c2c88d40aa46a90a9083c38271445

File: app/api/api_v1/schemas/document.py [hunk 2]
```diff
@@ -52,21 +52,22 @@ class FamilyDocumentResponse(BaseModel):
...
+    md5_sum: Optional[str] = None
+    cdn_object: Optional[str] = None
+    source_url: Optional[str] = None
+    content_type: Optional[str] = None
     # TODO: Remove after transition to multiple languages
     language: str
     languages: Sequence[str]
-    document_type: Optional[str]
-    document_role: Optional[str]
+    document_type: Optional[str] = None
```
Comment (diversemix) [inline app/api/api_v1/schemas/document.py:66]: Please note: https://linear.app/climate-policy-radar/issue/PDCT-248/schema-change-for-document-type ... this was raised after a conversation with @alaniswright ... This should not be optional in the future

File: pyproject.toml [hunk 3]
```diff
@@ -10,8 +10,8 @@ alembic = "^1.7.6"
 Authlib = "^0.15.5"
 bcrypt = "^3.2.0"
 boto3 = "^1.26"
-cpr-data-access = {git = "https://github.com/climatepolicyradar/data-access.git", tag = "v0.2.9"}
-fastapi = "^0.89.0"
+cpr-data-access = {git = "https://github.com/climatepolicyradar/data-access.git", rev = "4ec055db7cd93f2c9b275c5a6fd90145c6aa6851", extras = ["vespa"]}
```
Comment (diversemix) [inline pyproject.toml:13]: Why are we using `rev` and not `tag`?

File: app/api/api_v1/schemas/document.py (hunk 2 above)
Comment (olaughter) [reply app/api/api_v1/schemas/document.py:66]: Good spot, probably a separate change though?

File: pyproject.toml (hunk 3 above)
Comment (olaughter) [reply pyproject.toml:13]: The commit hasn't actually been merged yet, its part of: https://github.com/climatepolicyradar/data-access/pull/105 We wanted to test that the change in that PR worked here, I'd want it properly merged and switched here to a tag before we update prod though

**Review by diversemix — APPROVED** [review]

File: pyproject.toml (hunk 3 above)
Comment (olaughter) [reply pyproject.toml:13]: No problem, I'll tag first

**Pull request merged**